### ISSUE TITLE
Remove ESLint overrides once SDK fixes `any` issues

### ISSUE THREAD

**Issue by reefdog** [PhilanthropyDataCommons/front-end#739]
In #719, we convert some locally-defined types to use those provided by the SDK. Upon doing so, we discovered some shortcomings of the SDK (https://github.com/PhilanthropyDataCommons/service/issues/1004, ~https://github.com/PhilanthropyDataCommons/service/issues/1011~) that required inline ESLint disabling.

Once those issues are resolved, let's be sure to remove the overrides.

🛑 This issue is currently blocked by:
- [ ] https://github.com/PhilanthropyDataCommons/service/issues/1004
- ~https://github.com/PhilanthropyDataCommons/service/issues/1011~ Scratch this; it's unrelated and a wontfix.
